crates/db_schema/replaceable_schema/triggers.sql

```diff
@@ -615,3 +615,98 @@ CREATE TRIGGER change_values
     FOR EACH ROW
     EXECUTE FUNCTION r.private_message_change_values ();
 
+-- When creating or resolving a report, update a report count
+-- on the post and comment aggregate tables
+CREATE FUNCTION r.update_post_aggregates_report_count ()
+    RETURNS TRIGGER
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    UPDATE
+        post_aggregates
+    SET
+        report_count = report_count + 1,
+        unresolved_report_count = unresolved_report_count + 1
+    WHERE
+        post_id = NEW.post_id;
+    RETURN NULL;
+END
+$$;
+
+CREATE TRIGGER post_report_aggregates
+    AFTER INSERT ON post_report
+    FOR EACH ROW
+    EXECUTE FUNCTION r.update_post_aggregates_report_count ();
+
+-- When resolving / unresolving a report, update the unresolved_report_count
+CREATE FUNCTION r.update_post_aggregates_unresolved_report_count ()
+    RETURNS TRIGGER
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    UPDATE
+        post_aggregates
+    SET
+        unresolved_report_count = unresolved_report_count + CASE WHEN NEW.resolved THEN
+            -1
+        ELSE
+            1
+        END
```

Comment:
If its resolved, then decrement that column. If it switches back to unresolved, then increment it.